### PR TITLE
Replace jaccard_similarity_score with jaccard_score

### PR DESCRIPTION
**Pull request by joanne-tsai**
jaccard_similarity_score was deprecated since version 0.21 by sklearn

### CODE REVIEW

File: tests/test_metrics_comparative.py [hunk 1]
```diff
@@ -29,12 +29,10 @@ def test_binary_jaccard_reference(truth_binary_values, prediction_binary_values)
     cm = create_binary_confusion_matrix(truth_binary_values, prediction_binary_values)
 
     value = metrics.binary_jaccard(cm)
-    # sklearn has a very idiosyncratic implementation of jaccard_similarity_score; unless the input
-    # arrays are wrapped in an additional dimension, the result is actually the accuracy score
-    # see: https://github.com/scikit-learn/scikit-learn/issues/3037
-    reference_value = sklearn.metrics.jaccard_similarity_score(
+    reference_value = sklearn.metrics.jaccard_score(
         np.expand_dims(truth_binary_values, axis=0),
```
Comment (brianhelba) [inline tests/test_metrics_comparative.py]: Do we still need to wrap these with [`np.expand_dims`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.expand_dims.html)? That was previously taking an array like `[1, 0, 1]` and wrapping it into `[[1, 0, 1]]`. Is the new implementation of `sklearn.metrics.jaccard_score` now able to take the arrays directly (like all the other `sklearn.metrics`)?

File: tests/test_metrics_comparative.py [hunk 2]
```diff
@@ -32,6 +32,7 @@ def test_binary_jaccard_reference(truth_binary_values, prediction_binary_values)
     reference_value = sklearn.metrics.jaccard_score(
         np.expand_dims(truth_binary_values, axis=0),
         np.expand_dims(prediction_binary_values, axis=0),
+        average='micro',
```
Comment (brianhelba) [inline tests/test_metrics_comparative.py]: Since this is a binary classification, we shouldn't need to think about the averaging strategy. For what it's worth, you can read up on micro vs macro here: https://datascience.stackexchange.com/a/16001 . Note, most of our "aggregrate" metrics use a macro averaging strategy, except for the balanced multiclass accuracy, which uses micro averaging (the *micro* is what makes it *balanced*).

In this case, does the need to use `micro` go away if we get rid of the extra array wrapping?

**Review by brianhelba — APPROVED** [review]
Great!
